website/src/theme/BlogSidebar/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useEffect, useState } from 'react';
+import clsx from 'clsx';
+import Link from '@docusaurus/Link';
+import styles from './styles.module.css';
+import { useHistory } from '@docusaurus/router';
+
+export default function BlogSidebar({count}) {
+  const [ selected, setSelected ] = useState();
+  const history = useHistory();
+  const path = history.location.pathname.split('/');
+
+  useEffect(() => {
+    if (path.length === 2) {
+      setSelected('All');
+    } else if (path.length === 4) {
+      setSelected(path[3]);
+    } else {
+      setSelected('All');
+    }
+  }, [path]);
+
+  if (!count) {
+    return null;
+  }
+
+  const handleTagClick = (tag) => {
+    setSelected(tag);
+    if (tag === "All") {
+      history.push('/blog');
+    } else {
+      if (tag.indexOf(' ') !== -1) {
+        tag = tag.replace(' ', '-');
+      }
+      history.push(`/blog/tags/${tag}`);
+    }
+  };
+
+  return (
+    <div className={clsx(styles.sidebar, 'thin-scrollbar')}>
+      <h3 className={styles.sidebarItemTitle}>Tags</h3>
+      <div className={styles.sidebarItemList}>
+        {Object.entries(count).map(([tag, num]) => (
+          <div
+            key={tag}
+            className={`${styles.sidebarItem} ${selected === tag ? styles.selected : ''}`}
```

Comment:
Here, you might want to check the comparison `selected === tag` especially for the `practical case` tag.
As, when clicking on practical case tag, it's background color doesn't hold. 
My guess for this bug is, you are comparing "practical case" and "practical-case" or something similar here.

Reply:
`practical case` has something special. I can’t directly set the route according to it, I need to set it as `practical case`, So when I read the state from the route, I forgot to restore it. This is a bug.